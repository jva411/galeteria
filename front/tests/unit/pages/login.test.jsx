/* eslint-disable no-undef */
import React from 'react'
import Login from '../../../pages/login'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'

describe('unit tests of page \'login\'', () => {

    test('should render lenovo logo', () => {
        render(<Login />)

        expect(screen.getByAltText(/Lenovo Logo/iu)).toBeInTheDocument()
    })

    test('should render lenovo copyright', () => {
        render(<Login />)

        expect(screen.getByText(/Â© 2019-2021 Lenovo Sentiment. All rights reserved./iu))
    })

    test('should render input for username and password', () => {
        render(<Login />)

        expect(screen.getByPlaceholderText(/username/iu)).toBeInTheDocument()
        expect(screen.getByPlaceholderText(/password/iu)).toBeInTheDocument()
    })

    test('should render button to login', () => {
        render(<Login />)

        const button = screen.getByRole('button')

        expect(button).toBeInTheDocument()
        expect(button).toHaveTextContent(/login/iu)
    })

    test('should Index match with Snapshot', () => {
        const tree = renderer.create(<Login />).toJSON()

        expect(tree).toMatchSnapshot()
    })
})
