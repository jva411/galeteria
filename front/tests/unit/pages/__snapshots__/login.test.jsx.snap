// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit tests of page 'login' should Index match with Snapshot 1`] = `
<div
  className="css-1atv9zg"
>
  <div
    className="css-eni4rp"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDg5IiBoZWlnaHQ9IjEwMyIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <noscript>
        <img
          alt="Lenovo Logo"
          decoding="async"
          src="/logo2.png"
          srcSet="/_next/image?url=%2Flogo2.png&w=640&q=75 1x, /_next/image?url=%2Flogo2.png&w=1080&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
      <img
        alt="Lenovo Logo"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <div
    className="css-pw8lt9"
  >
    <div
      className="chakra-stack css-owjkmg"
    >
      <p>
        LENOVO SENTIMENT
      </p>
      <p>
        Welcome! Please login to your account.
      </p>
      <form
        onSubmit={[Function]}
      >
        <div
          className="labeled-input-div"
        >
          <input
            id="username"
            name="username"
            onChange={[Function]}
            placeholder="Username"
            type="text"
            value=""
          />
          <label
            htmlFor="username"
          >
            Username
          </label>
        </div>
        <div
          className="labeled-input-div"
        >
          <input
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            style={
              Object {
                "paddingRight": "5rem",
              }
            }
            type="password"
            value=""
          />
          <label
            htmlFor="password"
          >
            Password
          </label>
          <svg
            className="chakra-icon chakra-icon css-1kgncjz"
            focusable={false}
            onClick={[Function]}
            title="password-title"
            viewBox="0 0 24 24"
          >
            <g
              fill="currentColor"
            >
              <path
                d="M23.2,10.549a20.954,20.954,0,0,0-4.3-3.6l4-3.995a1,1,0,1,0-1.414-1.414l-.018.018a.737.737,0,0,1-.173.291l-19.5,19.5c-.008.007-.018.009-.026.017a1,1,0,0,0,1.631,1.088l4.146-4.146a11.26,11.26,0,0,0,4.31.939h.3c4.256,0,8.489-2.984,11.051-5.8A2.171,2.171,0,0,0,23.2,10.549ZM16.313,13.27a4.581,4.581,0,0,1-3,3.028,4.3,4.3,0,0,1-3.1-.19.253.253,0,0,1-.068-.407l5.56-5.559a.252.252,0,0,1,.407.067A4.3,4.3,0,0,1,16.313,13.27Z"
              />
              <path
                d="M7.615,13.4a.244.244,0,0,0,.061-.24A4.315,4.315,0,0,1,7.5,12,4.5,4.5,0,0,1,12,7.5a4.276,4.276,0,0,1,1.16.173.244.244,0,0,0,.24-.062l1.941-1.942a.254.254,0,0,0-.1-.421A10.413,10.413,0,0,0,12,4.75C7.7,4.692,3.4,7.7.813,10.549a2.15,2.15,0,0,0-.007,2.9,21.209,21.209,0,0,0,3.438,3.03.256.256,0,0,0,.326-.029Z"
              />
            </g>
          </svg>
        </div>
        <button
          type="submit"
        >
          Login
        </button>
      </form>
      <p>
        Â© 2019-2021 Lenovo Sentiment. All rights reserved.
      </p>
    </div>
  </div>
</div>
`;
